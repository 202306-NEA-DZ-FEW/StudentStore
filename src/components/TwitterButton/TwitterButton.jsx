import { auth, db } from "@/util/firebase";
import { TwitterAuthProvider, signInWithPopup } from "firebase/auth";
import { doc, setDoc } from "firebase/firestore";
import { useRouter } from "next/router";
import { useEffect } from "react";
import { BsTwitter } from "react-icons/bs";
import { ToastContainer, toast } from "react-toastify";
import { twMerge } from "tailwind-merge";

export default function TwitterButton({ children, className }) {
    useEffect(() => {
        return () => {
            toast.dismiss();
        };
    }, []);
    const route = useRouter();
    function signInWithTwitter() {
        const provider = new TwitterAuthProvider();
        signInWithPopup(auth, provider)
            .then((cred) => {
                console.log(cred);
                const user = cred.user;
                const userRef = doc(db, "userinfo", user.uid);
                toast.loading("Please wait");
                setDoc(userRef, {
                    name: user.displayName,
                    surname: "",
                    email: user.email,
                    password: "",
                    school: "",
                    photo: user.photoURL,
                    phoneNumber: user.phoneNumber,
                    gender: "",
                    country: "",
                    city: "",
                    zipcode: "",
                }).then(() => {
                    route.push("/home");
                });
            })
            .catch((error) => {
                console.log(error);
            });
    }
    return (
        <div>
            <button
                className={twMerge(
                    "border-[#28C7FA] text-[#28C7FA] bg-white border p-1 flex gap-1 items-center rounded-[22px] hover:shadow-lg hover:shadow-[rgba(50,49,77,0.3)]",
                    className
                )}
                onClick={signInWithTwitter}
            >
                <BsTwitter className='text-2xl ml-1' />
                <p className='text-md mx-1 font-semibold lg:mx-2 lg:mr-4'>
                    {children}
                </p>
            </button>
            <ToastContainer />
        </div>
    );
}
