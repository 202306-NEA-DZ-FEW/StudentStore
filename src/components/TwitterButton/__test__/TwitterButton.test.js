import renderer from "react-test-renderer";
import TwitterButton from "../TwitterButton";

jest.mock("../../../util/firebase", () => {
    return {
        initializeApp: jest.fn(),
    };
});

it("renders correctly", () => {
    const tree = renderer.create(<TwitterButton />).toJSON();
    expect(tree).toMatchSnapshot();
});
