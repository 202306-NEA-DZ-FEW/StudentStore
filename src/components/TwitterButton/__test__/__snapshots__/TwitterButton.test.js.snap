// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    className="border-[#28C7FA] text-[#28C7FA] bg-white border p-1 flex gap-1 items-center rounded-[22px] hover:shadow-lg hover:shadow-[rgba(50,49,77,0.3)]"
    onClick={[Function]}
  >
    <svg
      className="text-2xl ml-1"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M389.2 48h70.6L305.6 224.2 487 464H345L233.7 318.6 106.5 464H35.8L200.7 275.5 26.8 48H172.4L272.9 180.9 389.2 48zM364.4 421.8h39.1L151.1 88h-42L364.4 421.8z"
      />
    </svg>
    <p
      className="text-md mx-1 font-semibold lg:mx-2 lg:mr-4"
    />
  </button>
  <div
    className="Toastify"
  />
</div>
`;
